src/lambdas/nextjs-bucket-deployment.ts

```diff
@@ -272,19 +278,22 @@ function uploadObjects({
   baseLocalDir: string;
   putConfig: CustomResourceProperties['putConfig'];
 }) {
-  const putObjectInputs: PutObjectCommandInput[] = filePaths.map((path) => {
-    const contentType = mime.lookup(path) || undefined;
-    const putObjectOptions = getPutObjectOptions({ path, putConfig });
-    const key = createS3Key({ keyPrefix, path, baseLocalDir });
-    return {
-      ContentType: contentType,
-      ...putObjectOptions,
-      Bucket: bucket,
-      Key: key,
-      Body: createReadStream(path),
-    };
-  });
-  return Promise.all(putObjectInputs.map((input) => s3.send(new PutObjectCommand(input))));
+  for await (const filePathChunk of chunkArray(filePaths, 10)) {
```

Comment:
@jadenv, any particular reason you chose `10`? Why not `50` or `100`? If we're using 1536 MB Lambda, then 100 would mean we're assuming files are less than 1536 / 100 = 15MB. Do you think that's a fair assumption? Not sure what average image/asset size is. Plus, I know code would take up some memory so it's probably something more like (1536-500) / 100 = 10MB. What do you think?

Reply:
@bestickley good question, no particular reason I choose 10 other than just trying to get it to work. I think your calculation for 100 seems reasonable, I will change it to that!